pkgs/development/web/publii/default.nix

```diff
@@ -0,0 +1,109 @@
+{ stdenv
+, lib
+, fetchurl
+, autoPatchelfHook
+, makeShellWrapper
+, wrapGAppsHook
+, alsa-lib
+, at-spi2-atk
+, at-spi2-core
+, atk
+, cairo
+, cups
+, dbus
+, expat
+, glib
+, gtk3
+, libsecret
+, mesa
+, nss
+, pango
+, udev
+, xdg-utils
+, xorg
+}: let
+  intName = "Publii";
+  version = "0.42.0";
+
+in stdenv.mkDerivation {
+  pname = lib.strings.toLower intName;
+  inherit version;
+
+  src = fetchurl {
+    url = "https://getpublii.com/download/Publii-${version}.deb";
+    hash = "sha256-ohwoKhLT75gOC4f6R1tjhSygNw5lR7kbWBG6qpcP368=";
+  };
+
+  dontConfigure = true;
+  dontBuild = true;
+  dontWrapGApps = true;
+
+  nativeBuildInputs = [
+    autoPatchelfHook
+    makeShellWrapper
+    wrapGAppsHook
+  ];
+
+  buildInputs = [
+    alsa-lib
+    at-spi2-atk
+    at-spi2-core
+    atk
+    cairo
+    cups
+    dbus
+    expat
+    glib
+    gtk3
+    libsecret
+    mesa
+    nss
+    pango
+    xorg.libX11
+    xorg.libxcb
+  ];
+
+  unpackPhase = ''
+    ar p $src data.tar.xz | tar xJ
+  '';
+
+  installPhase = ''
+    runHook preInstall
+
+    mkdir -p $out/bin
+
+    mv usr/share $out
+    substituteInPlace $out/share/applications/${intName}.desktop \
+      --replace 'Exec=/opt/Publii/Publii' 'Exec=${intName}'
+
+    mv opt $out
+
+    runHook postInstall
+  '';
+
+  preFixup = ''
+    # makeWrapper defaults to makeBinaryWrapper due to wrapGAppsHook
+    # but we need a shell wrapper specifically for xdg-fix. Make xdg-open overrideable at runtime.
+    # Electron is trying to open udev via dlopen()
+    # and for some reason that doesn't seem to be impacted from the rpath.
+    # Adding udev to LD_LIBRARY_PATH fixes that.
+    makeShellWrapper $out/opt/${intName}/${intName} $out/bin/${intName} \
+      "''${gappsWrapperArgs[@]}" \
+      --suffix PATH : ${lib.makeBinPath [ xdg-utils ]} \
```

Comment:
This should also work with the binary wrapper

Reply:
just changing to makeBinaryWrapper or what is it about?

Reply:
This line should also work with makeBinaryWrapper and makeGapssWrapper. Not sure what issue you encountered in the comment above but it shouldn't happen.

Reply:
I got it partially from here: https://github.com/NixOS/nixpkgs/blob/d69661180de00d7284bc1503fa7b61769c82309d/pkgs/applications/misc/1password-gui/linux.nix#L123 and as the errors were similar / the software-stack (electron) is I tried that.

Trying atm. to change it and check if its working/how.